wordpress-mobile/WordPress-Android
# Add view model to photo picker

This PR prepares the Photo picker for the media picker consolidation. I've transformed the fragment to use a ViewModel and a UI state object with the list of media items. This moves the screen towards a unidirectional data flow and makes it less error prone are more readable. There should be no functionality changes.

To test:
- Go to a Gutenberg post
- Add an image block
- Click on "Add image"
- Select "Choose from device"
- The photo picker should behave the same way as before
- The screen keeps position on orientation
- The animations still work correctly
- Adding image to a block works correctly

To test 2:
- Repeat steps from the previous test with "Gallery" block

To test 3:
- Repeat steps with a "Video" block

PR submission checklist:
